packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +634,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
+): [resultValue: string, updatedChange: string, htmlIndex: number, plainTextSelectionEndIndex: number | null] => {
+  if (tag.tagType !== 'msft-mention' || tag.plainTextBeginIndex === undefined) {
+    return ['', processedChange, lastProcessedHTMLIndex, null];
+  }
+  let result = '';
+  let plainTextSelectionEndIndex: number | null = null;
+  let rangeStart: number;
+  let rangeEnd: number;
+  // check if space symbol is handled in case if string looks like '<1 2 3>'
+  let isSpaceLengthHandled = false;
+  rangeStart = oldPlainText.lastIndexOf(' ', startIndex);
+  if (rangeStart !== -1 && rangeStart !== undefined && rangeStart > tag.plainTextBeginIndex) {
+    isSpaceLengthHandled = true;
+  }
+  rangeEnd = oldPlainText.indexOf(' ', oldPlainTextEndIndex);
+  if (rangeEnd === -1 || rangeEnd === undefined) {
+    // check if space symbol is not found
+    rangeEnd = plainTextEndIndex;
+  } else if (!isSpaceLengthHandled) {
+    // +1 to include the space symbol
+    rangeEnd += 1;
+  }
+  isSpaceLengthHandled = true;
+
+  if (rangeStart === -1 || rangeStart === undefined || rangeStart < tag.plainTextBeginIndex) {
+    rangeStart = tag.plainTextBeginIndex;
+  }
+  if (rangeEnd > plainTextEndIndex) {
+    rangeEnd = plainTextEndIndex;
+  }
+
+  if (rangeStart === tag.plainTextBeginIndex && rangeEnd === plainTextEndIndex) {
+    // the whole tag should be removed
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIdx) + processedChange;
+    plainTextSelectionEndIndex = tag.plainTextBeginIndex + processedChange.length;
+    processedChange = '';
+    lastProcessedHTMLIndex = closeTagIdx + closeTagLength;
+  } else {
+    // only part of the tag should be removed
+    let startChangeDiff = 0;
+    let endChangeDiff = 0;
+    // need to check only rangeStart > tag.plainTextBeginIndex as when rangeStart === tag.plainTextBeginIndex startChangeDiff = 0 and mentionTagLength shouldn't be subtracted
+    if (rangeStart > tag.plainTextBeginIndex) {
+      startChangeDiff = rangeStart - tag.plainTextBeginIndex - mentionTagLength;
+    }
+    endChangeDiff = rangeEnd - tag.plainTextBeginIndex - mentionTagLength;
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIdx + tag.openTagBody.length + startChangeDiff);
+    // plainTextSelectionEndIndex = rangeStart + processedChange.length;
+    lastProcessedHTMLIndex = tag.openTagIdx + tag.openTagBody.length + endChangeDiff;
+    // processed change should not be changed as it should be added after the tag
+  }
+  return [result, processedChange, lastProcessedHTMLIndex, plainTextSelectionEndIndex];
+};
+
+/* @conditional-compile-remove(mention) */
+/**
+ * Go through the text and update it with the changed text
+ *
+ * @private
+ */
+const updateHTML = (
+  htmlText: string,
+  oldPlainText: string,
+  newPlainText: string,
+  tags: TagData[],
+  startIndex: number,
+  oldPlainTextEndIndex: number,
+  change: string,
+  mentionTrigger: string
+): [string, number | null] => {
```

Comment:
Either in this PR or a further work item, use object instead of tuple
the reader just simply can't understand what does the second number mean when not looking deeply into the function

Reply:
Added to refactor task